execution/src/posix/execution.cc

```diff
@@ -314,9 +314,13 @@ namespace leatherman { namespace execution {
         }
 
         // Close all open file descriptors above stderr
+#if defined(__FreeBSD__)
```

Comment:
Instead of testing the operating system, we can tell CMake to check whether `closefrom(2)` exist or not in libc.  This would allow leatherman to use this system call when available (at least FreeBSD and OpenBSD have it), and rely on the for loop otherwise.

Adding the following to `execution/CMakeLists.txt` defines `HAS_CLOSEFROM` accordingly:

```cmake
check_library_exists(c closefrom /lib CLOSEFROM_IN_LIBC)
if ("${CLOSEFROM_IN_LIBC}" STREQUAL "1")
    add_compile_definitions(HAS_CLOSEFROM)
endif()
```

Reply:
Yeah, I think I'd prefer to see the detection done in CMake if possible. It looks like Solaris and other BSDs also have the call, so we should use it wherever it's available.